app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -335,8 +335,9 @@ class HomeFragment : Fragment() {
                 }
             }
         }, owner = this)
+
         if (context.settings().showPrivateModeContextualFeatureRecommender &&
-            browsingModeManager.mode.isPrivate) {
+            browsingModeManager.mode.isPrivate && !PrivateShortcutCreateManager.doesPrivateBrowsingPinnedShortcutExist(requireContext())) {
```

Comment:
nit: There's already a not-null context variable here (used for `context.settings()`)
```suggestion
            browsingModeManager.mode.isPrivate && !PrivateShortcutCreateManager.doesPrivateBrowsingPinnedShortcutExist(context)) {
```

Reply:
This line might be too long, check the linter